src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx

```diff
@@ -89,11 +89,18 @@ export default class OrganizationDiscover extends React.Component {
 
     // If there are aggregations, get data for chart
     if (queryBuilder.getInternal().aggregations.length > 0) {
+      const field =
+        queryBuilder.getExternal().fields.length > 0
+          ? queryBuilder.getExternal().fields[0]
+          : null;
+      const groupBy = field ? [field, 'time'] : ['time'];
+
       const chartQuery = {
         ...queryBuilder.getExternal(),
-        groupby: ['time'],
+        groupby: groupBy,
         rollup: 60 * 60 * 24,
-        orderby: 'time',
+        orderby: '-count',
+        limit: 15,
```

Comment:
Why 15? Isn't this too small? If we are fetching multiple data sets each broken down by day we could be missing data

Reply:
What number do you suggest? From a design perspective, we were worried about the chart looking muddy.

Reply:
If I understand correctly, the limit here is on the number of distinct time or field-time (depending on query) values. So if we query for 1 day, then this'll be the top 15 values for whatever field is for that day. If we query for 90 days we're still only ever going to return 15 data points over 15 or less days. It also doesn't guarantee that the returned values are part of the same series. I think a workaround could be to request the max, then filter it before rendering the chart, but it probably still won't be guaranteed.